connectors/es/sink.py

```diff
@@ -171,6 +180,43 @@ async def _bulk_api_call():
 
         return res
 
+    async def _log_bulk_operations(self, res):
+        for item in res.get("items", []):
+            if "index" in item:
+                action_item = "index"
+            elif "delete" in item:
+                action_item = "delete"
+            elif "create" in item:
+                action_item = "create"
+            elif "update" in item:
+                action_item = "update"
+            else:
+                # Should only happen, if the _bulk API changes
+                # Unlikely, but as this functionality could be used for audits we want to detect changes fast
+                self._logger.error(
+                    f"Unknown action item returned from _bulk API for item {item}"
+                )
+                return
+
+            doc_id = item[action_item].get("_id")
+            if doc_id is None:
+                # Should only happen, if the _bulk API changes
+                # Unlikely, but as this functionality could be used for audits we want to detect changes fast
+                self._logger.error(f"Could not retrieve '_id' for item {item}")
+                return
+
+            result = item[action_item].get("result")
+            operation_failed = result not in SUCCESSFUL_RESULTS
+
+            if operation_failed:
```

Comment:
For failed operation, you should check 
```python
if "error" in item[action_item]
```

Reply:
See comment above. Maybe it makes sense to name the condition `non_successful_result`?